unreal_asset/src/lib.rs

```diff
@@ -796,8 +796,8 @@ impl<'a> Asset {
     }
 
     pub fn add_fname(&mut self, slice: &str) -> FName {
-        let name = FName::from_slice(slice);
-        self.add_name_reference(name.content.clone(), false);
+        let mut name = FName::from_slice(slice);
+        name.index = self.add_name_reference(name.content.clone(), false);
```

Comment:
The `add_fname` function shouldn't set the index of fname to `add_name_reference` result, because the `index` field is not the index in the name map but rather the "duplicaion" index.

Here's an example: 
FName(content: "Something", index: 0) results in unreal string "Something_0"
FName(content: "Something", index: 1) results in unreal string "Something_1"